lib/Remotery.h

```diff
@@ -358,11 +383,49 @@ typedef enum rmtSampleFlags
     RMT_OPTIONAL(RMT_ENABLED, _rmt_EndCPUSample())
 
 
+#define rmt_IterateChildren(iter, sample)                                           \
+    RMT_OPTIONAL(RMT_ENABLED, _rmt_IterateChildren(iter, sample))
```

Comment:
Tweaked the api from
```
rmtSampleIterator iter = rmt_IterateChildren(sample);
```
to
```
rmtSampleIterator iter;
rmt_IterateChildren(&iter, sample);
```
in order to both make it more C89 compatible, and also make it work with RMT_OPTIONAL better.